include/seqcmd.h

```diff
@@ -0,0 +1,652 @@
+#ifndef SEQCMD_H
+#define SEQCMD_H
+
+// ==== Primary commands ====
+
+#define SEQ_CMD_MASK 0xF0000000
+
+typedef enum {
+    /* 0x0 */ SEQ_CMD_PLAY,
+    /* 0x1 */ SEQ_CMD_STOP,
+    /* 0x2 */ SEQ_CMD_QUEUE,
+    /* 0x3 */ SEQ_CMD_UNQUEUE,
+    /* 0x4 */ SEQ_CMD_SET_PLAYER_VOLUME,
+    /* 0x5 */ SEQ_CMD_SET_PLAYER_FREQ,
+    /* 0x6 */ SEQ_CMD_SET_CHANNEL_VOLUME,
+    /* 0x7 */ SEQ_CMD_SET_PLAYER_IO,
+    /* 0x8 */ SEQ_CMD_SET_CHANNEL_IO,
+    /* 0x9 */ SEQ_CMD_DISABLE_CHANNEL_IO,
+    /* 0xA */ SEQ_CMD_DISABLE_CHANNELS,
+    /* 0xB */ SEQ_CMD_TEMPO_CMD,
+    /* 0xC */ SEQ_CMD_SETUP_CMD,
+    /* 0xD */ SEQ_CMD_SET_CHANNEL_FREQ,
+    /* 0xE */ SEQ_CMD_GLOBAL_CMD,
+    /* 0xF */ SEQ_CMD_RESET_HEAP
+} SeqCmdType;
+
+// ==== Secondary commands ====
+
+// Subset of SEQ_CMD_TEMPO_CMD
+typedef enum {
+    /* 0x0 */ SEQ_SUB_CMD_TEMPO_SET,
+    /* 0x1 */ SEQ_SUB_CMD_TEMPO_SPEED_UP,
+    /* 0x2 */ SEQ_SUB_CMD_TEMPO_SLOW_DOWN,
+    /* 0x3 */ SEQ_SUB_CMD_TEMPO_SCALE,
+    /* 0x4 */ SEQ_SUB_CMD_TEMPO_RESET
+} TempoCmdType;
+
+// Subset of SEQ_CMD_SETUP_CMD
+typedef enum {
+    /* 0x0 */ SEQ_SUB_CMD_SETUP_RESTORE_VOLUME,
+    /* 0x1 */ SEQ_SUB_CMD_SETUP_SEQ_UNQUEUE,
+    /* 0x2 */ SEQ_SUB_CMD_SETUP_RESTART_SEQ,
+    /* 0x3 */ SEQ_SUB_CMD_SETUP_TEMPO_SCALE,
+    /* 0x4 */ SEQ_SUB_CMD_SETUP_TEMPO_RESET,
+    /* 0x5 */ SEQ_SUB_CMD_SETUP_PLAY_SEQ,
+    /* 0x6 */ SEQ_SUB_CMD_SETUP_SET_FADE_TIMER,
+    /* 0x7 */ SEQ_SUB_CMD_SETUP_RESTORE_VOLUME_IF_QUEUED,
+    /* 0x8 */ SEQ_SUB_CMD_SETUP_RESTORE_VOLUME_WITH_SCALE,
+    /* 0x9 */ SEQ_SUB_CMD_SETUP_DISABLE_CHANNELS,
+    /* 0xA */ SEQ_SUB_CMD_SETUP_SET_PLAYER_FREQ,
+    /* 0xE */ SEQ_SUB_CMD_SETUP_POP_CACHE = 0xE,
+    /* 0xF */ SEQ_SUB_CMD_SETUP_RESET_SETUP_CMDS
+} SetupCmdType;
+
+// Subset of SEQ_CMD_GLOBAL_CMD
+typedef enum {
+    /* 0x0 */ SEQ_SUB_CMD_GLOBAL_SET_SOUND_MODE,
+    /* 0x1 */ SEQ_SUB_CMD_GLOBAL_DISABLE_NEW_SEQUENCES
+} SubCmdType;
+
+// ==== Audio Sequence Primary Commands ====
+
+/**
+ * ARGS
+ *   playerIndex (p), fadeInTimer (t), seqArgs (a), seqId (s)
+ *
+ * FORMAT
+ *   Capital U is unused
+ *   0pttaass
```

Comment:
wondering if this format section on each of these comments is helpful. The structure is pretty easily determined by just looking at the shifts in the macro definition itself right?

Reply:
In most cases, yeah. There are some cases where a `(u16)` cast is needed when only a `(u8)` is read from i.e. 0x000000tt`, the casting was a bit weird in that it was a bit picky to get matching, and doesn't always line up with how variables are read from. So not all of these macro's are 100% safe.

I also leaned towards putting more info in to start, then we can always strip out the extraneous info later, so I'm not too worried if these go